### PR TITLE
Clean up: AddShrine() [Second attempt]

### PR DESCRIPTION
**Pull request by kphoenix137**

### CODE REVIEW

File: Source/objects.cpp [hunk 1]
```diff
@@ -1288,27 +1288,33 @@ void AddBarrel(Object &barrel)
 void AddShrine(Object &shrine)
 {
 	shrine._oRndSeed = AdvanceRndSeed();
-	bool slist[NumberOfShrineTypes];
-
 	shrine._oPreFlag = true;
 
-	int shrines = gbIsHellfire ? NumberOfShrineTypes : 26;
+	int shrineCount = gbIsHellfire ? NumberOfShrineTypes : 26;
+	bool slist[NumberOfShrineTypes] = { false };
```
Comment (qndel) [inline Source/objects.cpp:1294]: you can't do this

Comment (kphoenix137) [reply Source/objects.cpp:1294]: I'm not sure I understand what you mean?
`bool slist[NumberOfShrineTypes];` = correct
bool slist[NumberOfShrineTypes] = { false };` = incorrect ?
I'm just initializing it to false.

Comment (StephenCWills) [reply Source/objects.cpp:1294]: I think that by rearranging the code, you've tricked him into thinking you used `shrineCount` as the array size.

Comment (qndel) [reply Source/objects.cpp:1294]: > I think that by rearranging the code, you've tricked him into thinking you used `shrineCount` as the array size.

yup haha

Comment (qndel) [inline Source/objects.cpp:1294]:
```suggestion
	bool slist[NumberOfShrineTypes] = { };
```
{false} only sets first element to false

**Review by AJenbo — APPROVED** [review]

File: Source/objects.cpp [hunk 2]
```diff
@@ -1288,27 +1288,33 @@ void AddBarrel(Object &barrel)
 void AddShrine(Object &shrine)
 {
 	shrine._oRndSeed = AdvanceRndSeed();
-	bool slist[NumberOfShrineTypes];
-
 	shrine._oPreFlag = true;
 
-	int shrines = gbIsHellfire ? NumberOfShrineTypes : 26;
+	int shrineCount = gbIsHellfire ? NumberOfShrineTypes : 26;
+	bool slist[NumberOfShrineTypes] = { };
```
Comment (AJenbo) [inline Source/objects.cpp:1294]:
```suggestion
	bool slist[NumberOfShrineTypes] = {};
```

**Review by AJenbo — APPROVED** [review]

**Pull request merged**